pkgs/development/tools/rust/cargo-pgx/buildPgxExtension.nix

```
@@ -0,0 +1,177 @@
+
+let
+  maybeDebugFlag = if buildType != "release" then "--debug" else "";
+  maybeEnterBuildAndTestSubdir =
+    if buildAndTestSubdir != null then ''
+      export CARGO_TARGET_DIR="$(pwd)/target"
+      pushd ${buildAndTestSubdir}
+    '' else "";
+  maybeLeaveBuildAndTestSubdir = if buildAndTestSubdir != null then "popd" else "";
+
+  pgxPostgresMajor = builtins.head (lib.splitString "." postgresql.version);
+  preBuildAndTest = ''
+    export PGX_HOME=$(mktemp -d)
+    mkdir -p $PGX_HOME/${pgxPostgresMajor}
+
+    cp -r -L ${postgresql}/. $PGX_HOME/${pgxPostgresMajor}/
+    chmod -R ugo+w $PGX_HOME/${pgxPostgresMajor}
+    cp -r -L ${postgresql.lib}/lib/. $PGX_HOME/${pgxPostgresMajor}/lib/
+
+    ${cargo-pgx}/bin/cargo-pgx pgx init \
+      --pg${pgxPostgresMajor} $PGX_HOME/${pgxPostgresMajor}/bin/pg_config
+
+    # This is primarily for Mac or other Nix systems that don't use the nixbld user.
+    export USER=$(whoami)
+    export PGDATA=$PGX_HOME/data-${pgxPostgresMajor}/
+    export NIX_PGLIBDIR=$PGX_HOME/${pgxPostgresMajor}/lib
+
+    echo "unix_socket_directories = '$(mktemp -d)'" > $PGDATA/postgresql.conf
+    ${postgresql}/bin/pg_ctl start
+    ${postgresql}/bin/createuser -h localhost --superuser --createdb $USER || true
+    ${postgresql}/bin/pg_ctl stop
+  '';
+
+  argsForBuildRustPackage = builtins.removeAttrs args [ "postgresql" ];
+
+  # so we don't accidentally `(rustPlatform.buildRustPackage argsForBuildRustPackage) // { ... }` because
+  # we forgot parentheses
+  finalArgs = argsForBuildRustPackage // {
+    buildInputs = (args.buildInputs or [ ]) ++ [ postgresql ] ++ lib.optionals stdenv.isDarwin [ Security ];
+
+    nativeBuildInputs = (args.nativeBuildInputs or [ ]) ++ [
+      cargo-pgx
+      rustc
+      rustfmt
+      pkg-config
+      rustPlatform.bindgenHook
+    ];
+
+    postPatch = ''
+      ${args.postPatch or ""}
+      patchShebangs .
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+
+      echo "Executing cargo-pgx buildPhase"
+      ${preBuildAndTest}
+      ${maybeEnterBuildAndTestSubdir}
+
+      export NIX_PGLIBDIR=${postgresql.out}/lib
+
+      PGX_BUILD_FLAGS="--frozen -j $NIX_BUILD_CORES ${builtins.concatStringsSep " " cargoBuildFlags}" \
+      ${cargo-pgx}/bin/cargo-pgx pgx package \
+      --pg-config ${postgresql}/bin/pg_config \
+      ${maybeDebugFlag} \
+      --features "${builtins.concatStringsSep " " buildFeatures}" \
+      --out-dir $out
+
+      export NIX_PGLIBDIR=$PGX_HOME/${pgxPostgresMajor}/lib
+
+      ${maybeLeaveBuildAndTestSubdir}
+      echo "Finished cargo-pgx buildPhase"
+
+      runHook postBuild
+    '';
+
+    preCheck = ''
+      ${preBuildAndTest}
+      ${args.preCheck or ""}
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      echo "Executing buildPgxExtension install"
+
+      ${maybeEnterBuildAndTestSubdir}
+
+      ${cargo-pgx}/bin/cargo-pgx pgx stop all
+
+      mv -v $out/${postgresql.out}/* $out
+      rm -rfv $out/nix
```

```suggestion
      mv $out/${postgresql.out}/* $out
      rm -rf $out/nix
```
We don't need to spam the log.